apps/worker/src/app/workflow/usecases/send-message/send-message-chat.usecase.ts

```diff
@@ -71,11 +74,17 @@ export class SendMessageChat extends SendMessageBase {
     Sentry.addBreadcrumb({
       message: 'Sending Chat',
     });
-    const chatChannel: NotificationStepEntity = command.step;
-    if (!chatChannel?.template) throw new PlatformException('Chat channel template not found');
+    const step: NotificationStepEntity = command.step;
+    if (!step?.template) throw new PlatformException('Chat channel template not found');
 
     const tenant = await this.handleTenantExecution(command.job);
 
+    const template = await this.processVariants(command, tenant, subscriber, command.payload);
+
+    if (template) {
+      step.template = template;
+    }
```

Comment:
here we check if there is a variant, if it exists we will override the template.